packages/esm-ward-app/src/ward-patient-card/row-elements/ward-patient-identifier.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { type WardPatientCardElement } from '../../types';
+import { type PatientCardElementConfig } from '../../config-schema';
+import { Tag } from '@carbon/react';
+
+const identifierCompareFunction = (patientIdentifier1, patientIdentifier2) => {
+  const sortByDate =
+    (!patientIdentifier1.preferred && !patientIdentifier2.preferred) ||
+    (patientIdentifier1.preferred && patientIdentifier2.preferred);
+  if (sortByDate) {
+    const patientIdentifier2Date = new Date(
+      patientIdentifier2.auditInfo.dateChanged || patientIdentifier2.auditInfo.dateCreated,
+    );
+    const patientIdentifier1Date = new Date(
+      patientIdentifier1.auditInfo.dateChanged || patientIdentifier1.auditInfo.dateCreated,
+    );
+    return patientIdentifier2Date.valueOf() - patientIdentifier1Date.valueOf();
+  }
+  if (patientIdentifier1.preferred) return -1;
+  if (patientIdentifier2.preferred) return 1;
+  return 0;
+};
```

Comment:
- Let's add typing as much as we can when TypeScript can't infer it.
- Add a comment on what this does. It's not obvious 
- use the newer (and better) `??` nullish coalescing operator over `||` 
- I prefer this pattern of "compare by A, if tie, then compare by B, if tie, then compare by C..."

```suggestion
const identifierCompareFunction = (pi1 : PatientIdentifier, pi2: PatientIdentifier) => {
  let comp = (pi1.preferred? 1 : 0) - (pi2.preferred? 1 : 0);
  if(comp == 0) {
    const date1 = pi1.auditInfo.dateChanged ?? pi1.auditInfo.dateCreated;
    const date2 = pi2.auditInfo.dateChanged ?? pi2.auditInfo.dateCreated;
    comp = date2.localeCompare(date1);
  }
  return comp;
}
```

Reply:
+1

Reply:
Made the change. Thanks @chibongho @mseaton .